src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/proposal/quickfix/RemoveParamAnnotationQuickFix.java

```diff
@@ -14,97 +14,128 @@
-                                     String label, List<String> annotationsToRemove) {
-
-        final PsiElement node = context.getCoveredNode();
-        final PsiClass parentType = getBinding(node);
-        final PsiMethod method = PsiTreeUtil.getParentOfType(node, PsiMethod.class);
-
-        final PsiParameter parameter = method.getParameterList().getParameter(parameterIndex);
-        final PsiAnnotation[] psiAnnotations = parameter.getAnnotations();
-        final List<PsiAnnotation> psiAnnotationsToRemove = new ArrayList<>();
-        Arrays.stream(psiAnnotations).forEach(a -> {
-            if (annotationsToRemove.stream().anyMatch(m -> m.equals(a.getQualifiedName()))) {
-                psiAnnotationsToRemove.add(a);
-            }
-        });
-
-        RemoveAnnotationsProposal proposal = new RemoveAnnotationsProposal(label, context.getSource().getCompilationUnit(),
-                context.getASTRoot(), parentType, 0, psiAnnotationsToRemove);
-        CodeAction codeAction = context.convertToCodeAction(proposal, diagnostic);
-        if (codeAction != null) {
-            codeActions.add(codeAction);
-        }
-    }
-
-    protected PsiClass getBinding(PsiElement node) {
-        return PsiTreeUtil.getParentOfType(node, PsiClass.class);
-    }
-
-    protected static String getShortName(String qualifiedName) {
-        final int i = qualifiedName.lastIndexOf('.');
-        if (i != -1) {
-            return qualifiedName.substring(i+1);
-        }
-        return qualifiedName;
-    }
-}
+public abstract class RemoveParamAnnotationQuickFix implements IJavaCodeActionParticipant {
+
+     private final String[] annotations;
+     private static final Logger LOGGER = Logger.getLogger(RemoveParamAnnotationQuickFix.class.getName());
+     private static final String ANNOTATION_TO_REMOVE = "annotationsToRemove";
+     private static final String INDEX_KEY = "paramIndex";
+
+     public RemoveParamAnnotationQuickFix(String... annotations) {
+         this.annotations = annotations;
+     }
+
+     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic) {
+
+         final PsiElement node = context.getCoveredNode();
+         final PsiMethod method = PsiTreeUtil.getParentOfType(node, PsiMethod.class);
+
+         final List<CodeAction> codeActions = new ArrayList<>();
+         final PsiParameterList parameters = method.getParameterList();
+         int parametersCount = parameters.getParametersCount();
+
+         for (int i = 0; i < parametersCount; ++i) {
+             final PsiParameter parameter = parameters.getParameter(i);
+             final PsiAnnotation[] psiAnnotations = parameter.getAnnotations();
+             final List<String> annotationsToRemove = new ArrayList<>();
+             // Search for annotations to remove from the current method parameter.
+             Arrays.stream(psiAnnotations).forEach(a -> {
```

Comment:
How can you remove the check for null here? "The methods in this class (Arrays) all throw a NullPointerException, if the specified array reference is null..."